frontend/src/components/common/NavBar.tsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+import {
+  Navbar,
+  NavbarBrand,
+  NavbarContent,
+  NavbarItem,
+  Dropdown,
+  DropdownTrigger,
+  DropdownItem,
+  DropdownMenu,
+  DropdownSection,
+  Button,
+} from "@nextui-org/react";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { CLIENT_ROUTES } from "@/common/constants";
+import PeerPrepLogo from "@/components/common/PeerPrepLogo";
+import ProfilePictureAvatar from "./ProfilePictureAvatar";
+import { UserService } from "@/helpers/user/api_wrappers";
+import { MatchingService } from "@/helpers/matching/api_wrappers";
+
+const NavBar = () => {
+  const router = useRouter();
+
+  const handleQuickMatchButtonPress = () => {
+    const preferences = UserService.getUserPreferences();
+    MatchingService.submitMatchPreferences(preferences);
+  };
+  const handleEditProfileButtonPress = () => {
+    router.push(CLIENT_ROUTES.PROFILE);
+  };
+  const handleLogoutButtonPress = () => {
+    router.push(CLIENT_ROUTES.LOGOUT);
+  };
+
+  return (
+    <Navbar className="bg-black justify-stretch" maxWidth="full" height="50px">
+      <NavbarBrand className="flex-grow-0">
+        <Link href={CLIENT_ROUTES.HOME}>
+          <PeerPrepLogo width="30px" height="30px" />
+        </Link>
+      </NavbarBrand>
+      <NavbarContent className="hidden sm:flex gap-4" justify="start">
+        <NavbarItem>
+          <Link className="text-light-blue" href={CLIENT_ROUTES.QUESTIONS}>
+            Questions
+          </Link>
+        </NavbarItem>
+      </NavbarContent>
+      <NavbarContent as="div" justify="end">
+        <Button
+          className="bg-yellow text-black h-[30px]"
+          onPress={handleQuickMatchButtonPress}
```

Comment:
This is interesting. Based on a quick reference from the internet, `onPress` means the event will be triggered when the user first presses the button. Compared to `onClick` which will only be triggered when the user presses and releases the button. I believe most buttons in web dev should use `onClick` instead of `onPress`? Open for discussion.

Reply:
Ah, I thought about this as well when I was writing the code. I tried both out but could not really notice the difference. Either is fine for me as long as it is standardised, so I can just refactor the code to be whatever we decide it to be. 👍